### DISCUSSION TITLE
enhanced-crm-container

### DISCUSSION THREAD

**Discussion by Mauricio-Nocce** [twilio-professional-services/flex-project-template#243] [General]
Can someone give me an example of how I configure the enhanced-crm-container, for each task to open a different url?

**Comment by jared-hunter** [comment]
@Mauricio-Nocce can you elaborate a little on what information in the task results in a different url?

Are we talking about parameters on a single domain or are we talking about different domains?

If the latter, are we talking about the task carrying the full url (probably unsafe) or is there an enumerated list of urls and the task identifies which url to use?

**Comment by dremin** [comment]
We just merged some improvements to that feature which allow you to specify a URL that can include task attributes. More details are available in the updated docs: https://twilio-professional-services.github.io/flex-project-template/feature-library/flex-v2/enhanced-crm-container

**Comment by Mauricio-Nocce** [comment]
Hi, Jared.
it would be a single domain, but the client has several rows, each row opens a different popup, I'm still a little new with this twilio product, I wanted to understand how the attributes would be in this case.
________________________________
De: Jared Hunter ***@***.***>
Enviado: terça-feira, 13 de junho de 2023 19:00
Para: twilio-professional-services/flex-project-template ***@***.***>
Cc: Mauricio-Nocce ***@***.***>; Mention ***@***.***>
Assunto: Re: [twilio-professional-services/flex-project-template] enhanced-crm-container (Discussion #243)


@Mauricio-Nocce<https://github.com/Mauricio-Nocce> can you elaborate a little on what information in the task results in a different url?

Are we talking about parameters on a single domain or are we talking about different domains?

If the latter, are we talking about the task carrying the full url (probably unsafe) or is there an enumerated list of urls and the task identifies which url to use?

**Reply by jared-hunter** [reply]
Hi 👋

I'm not sure what you mean by different rows in this context but as @dremin mentioned the latest enhancements to the feature allow you to use attribute replacement in the URL as part of the formation of the URL

Taken from the docs:

_Within the url, you may include task and/or worker attributes via template variable substitution. To do so, use the format {{task.attribute_name_goes_here}} or {{worker.attribute_name_goes_here}} within the URL. If the attribute exists, the template variable is replaced with the contents of the attribute. Otherwise, the variable is replaced with an empty string._

So for example if your task attributes contain a variable _department_. You can use it in the URL as so _https://www.bing.con/search?q={{task.department}}_

**Comment by Mauricio-Nocce** [comment]
I'll check it out, thanks for help.


________________________________
De: Jared Hunter ***@***.***>
Enviado: quinta-feira, 15 de junho de 2023 09:34
Para: twilio-professional-services/flex-project-template ***@***.***>
Cc: Mauricio-Nocce ***@***.***>; Mention ***@***.***>
Assunto: Re: [twilio-professional-services/flex-project-template] enhanced-crm-container (Discussion #243)


Hi 👋

I'm not sure what you mean by different rows in this context but as @dremin<https://github.com/dremin> mentioned the latest enhancements to the feature allow you to use attribute replacement in the URL as part of the formation of the URL

Taken from the docs:

Within the url, you may include task and/or worker attributes via template variable substitution. To do so, use the format {{task.attribute_name_goes_here}} or {{worker.attribute_name_goes_here}} within the URL. If the attribute exists, the template variable is replaced with the contents of the attribute. Otherwise, the variable is replaced with an empty string.

So for example if your task attributes contain a variable department. You can use it in the URL as so https://www.bing.con/search?q={{task.department}}<https://www.bing.con/search?q=%7B%7Btask.department%7D%7D>
